File: Bachelor/Semester4/Systems_for_Design_and_Implementation/Java/Repository/src/main/java/repository/UserDBRepository.java
package repository;

import domain.User;
import domain.validator.Validator;
import utils.JdbcUtils;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.Optional;

public class UserDBRepository extends AbstractDBRepository<Long,User> implements UserRepository{

    public UserDBRepository(JdbcUtils jdbcUtils, Validator<Long, User> validator) {
        super(jdbcUtils,validator);
    }

    @Override
    protected PreparedStatement getSaveStatement(Connection connection, User entity) throws SQLException {
        PreparedStatement statement = connection.prepareStatement("INSERT INTO users(username, password) VALUES (?,?);");
        statement.setString(1, entity.getUsername());
        statement.setString(2, entity.getPassword());
        return statement;
    }

    @Override
    protected PreparedStatement getDeleteStatement(Connection connection, Long aLong) throws SQLException {
        PreparedStatement statement = connection.prepareStatement("DELETE FROM users WHERE id=?;");
        statement.setLong(1, aLong);
        return statement;
    }

    @Override
    protected PreparedStatement getUpdateStatement(Connection connection, User entity) throws SQLException {
        PreparedStatement statement = connection.prepareStatement("UPDATE users SET username=?, password=? WHERE id=?;");
        statement.setString(1, entity.getUsername());
        statement.setString(2, entity.getPassword());
        statement.setLong(3, entity.getId());
        return statement;
    }

    @Override
    protected PreparedStatement getFindStatement(Connection connection, Long aLong) throws SQLException {
        PreparedStatement statement = connection.prepareStatement("SELECT * FROM users WHERE id=?;");
        statement.setLong(1, aLong);
        return statement;
    }

    @Override
    protected PreparedStatement getFindAllStatement(Connection connection) throws SQLException {
        return connection.prepareStatement("SELECT * FROM users;");
    }

    @Override
    protected User extractEntity(ResultSet resultSet) {
        logger.traceEntry();
        User result = null;
        try {
            Long id = resultSet.getLong("id");
            String name = resultSet.getString("username");
            String password = resultSet.getString("password");
            result = new User(id, name, password);
        }catch (SQLException e){
            logger.error(e);
            System.out.println("Error "+e);
        }
        logger.traceExit(result);
        return result;
    }

    @Override
    public Optional<User> findByUsernamePassword(String username, String password) {
        logger.traceEntry("Find username {}, {}", username, password);
        Optional<User> result = Optional.empty();
        try(Connection connection = jdbcUtils.getConnection()){
            try(PreparedStatement statement = connection.prepareStatement("SELECT * FROM users WHERE username=? AND password=?;")){
                statement.setString(1,username);
                statement.setString(2,password);
                try(ResultSet resultSet = statement.executeQuery()){
                    if(resultSet.next()){
                        result = Optional.of(extractEntity(resultSet));
                    }
                }
            }
        }catch (SQLException e){
            logger.error(e);
            System.out.println("Error "+e);
        }
        logger.traceExit(result);
        return result;
    }
}

File: Bachelor/Semester4/Systems_for_Design_and_Implementation/C#/ProiectCShMPP/Repository/repository/UserDbRepository.cs
using System;
using System.Data;
using ProiectCShMPP.domain;
using ProiectCShMPP.domain.validator;
using ProiectCShMPP.utils;

namespace ProiectCShMPP.repository
{
    public class UserDbRepository : AbstractDBRepository<long, User>, IUserRepository
    {
        public UserDbRepository(IValidator<long, User> validator) : base(validator)
        {
        }

        protected override IDbCommand GetSaveCommand(IDbConnection connection, User entity)
        {
            IDbCommand command = connection.CreateCommand();
            command.CommandText = "INSERT INTO users(username, password) VALUES (@username, @password);";

            IDbDataParameter parameterUsername = command.CreateParameter();
            parameterUsername.ParameterName = "@username";
            parameterUsername.Value = entity.Username;

            IDbDataParameter parameterPassword = command.CreateParameter();
            parameterPassword.ParameterName = "@password";
            parameterPassword.Value = entity.Password;

            command.Parameters.Add(parameterUsername);
            command.Parameters.Add(parameterPassword);
            return command;
        }

        protected override IDbCommand GetDeleteCommand(IDbConnection connection, long id)
        {
            IDbCommand command = connection.CreateCommand();
            command.CommandText = "DELETE FROM users WHERE id=@id;";

            IDataParameter parameter = command.CreateParameter();
            parameter.ParameterName = "@id";
            parameter.Value = id;

            command.Parameters.Add(parameter);
            return command;
        }

        protected override IDbCommand GetUpdateCommand(IDbConnection connection, User entity)
        {
            IDbCommand command = connection.CreateCommand();
            command.CommandText = "UPDATE users SET username = @username, password = @password WHERE id = @id;";

            IDataParameter parameter = command.CreateParameter();
            parameter.ParameterName = "@username";
            parameter.Value = entity.Username;
            command.Parameters.Add(parameter);

            parameter = command.CreateParameter();
            parameter.ParameterName = "@password";
            parameter.Value = entity.Password;
            command.Parameters.Add(parameter);
            return command;
        }

        protected override IDbCommand GetFindCommand(IDbConnection connection, long id)
        {
            IDbCommand command = connection.CreateCommand();
            command.CommandText = "SELECT * FROM users WHERE id=@id;";
            IDbDataParameter parameter = command.CreateParameter();
            parameter.ParameterName = "@id";
            parameter.Value = id;
            command.Parameters.Add(parameter);
            return command;
        }

        protected override IDbCommand GetFindAllCommand(IDbConnection connection)
        {
            IDbCommand command = connection.CreateCommand();
            command.CommandText = "SELECT * FROM users;";
            return command;
        }

        protected override User ExtractEntity(IDataReader reader)
        {
            _logger.InfoFormat("Entry Extract Entity");
            long id = (long)reader["id"];
            string name = (string) reader["username"];
            string password = (String) reader["password"];
            User result = new User(id)
            {
                Username = name,
                Password = password
            };
            _logger.InfoFormat("Extracted {0}", result);
            return result;
        }

        public User FindByUsernamePassword(string username, string password)
        {
            _logger.InfoFormat("Entry find user {0},{1}", username,password);
            User result = null;
            using (IDbConnection connection = DbUtils.GetConnection())
            {
                using (IDbCommand command = connection.CreateCommand())
                {
                    command.CommandText = "SELECT * FROM users WHERE username=@username AND password=@password;";
                    IDbDataParameter parameter = command.CreateParameter();
                    parameter.ParameterName = "@username";
                    parameter.Value = username;
                    command.Parameters.Add(parameter);

                    parameter = command.CreateParameter();
                    parameter.ParameterName = "@password";
                    parameter.Value = password;
                    command.Parameters.Add(parameter);

                    using (IDataReader reader = command.ExecuteReader())
                    {
                        if (reader.Read())
                            result = ExtractEntity(reader);
                    }
                }
            }
            _logger.InfoFormat("Found {0}", result);
            return result;
        }
    }
}